kuma/wiki/utils.py

```diff
@@ -59,3 +67,42 @@ def tidy_content(content):
         return tidied.decode('utf-8'), errors
     else:
         return content
+
+
+def analytics_user_counts(*revs):
+    """Given some document revision numbers, returns a dict matching those
+    with the number of users Google Analytics thinks has visited each revision.
+
+    """
+
+    scopes = ['https://www.googleapis.com/auth/analytics.readonly']
+    credentials = ServiceAccountCredentials.from_json_keyfile_dict(
+        json.loads(config.GOOGLE_ANALYTICS_CREDENTIALS), scopes=scopes)
+    http_auth = credentials.authorize(Http())
+    service = build('analyticsreporting', 'v4', http=http_auth)
+
+    request = service.reports().batchGet(
+        body={
+            'reportRequests': [
+                # `dimension12` is the custom variable containing a page's rev #.
+                {
+                    'dimensions': [{'name': 'ga:dimension12'}],
+                    'metrics': [{'expression': 'ga:users'}],
+                    'dimensionFilterClauses': [
+                        {
+                            'filters': [
+                                {'dimensionName': 'ga:dimension12',
+                                 'operator': 'IN_LIST',
+                                 'expressions': map(str, revs)}
+                            ]
+                        }
+                    ],
+                    'viewId': '66726481'  # PK of the developer.mozilla.org site on GA.
+                }
+            ]
+        })
+
+    response = request.execute()
+
+    return {int(row['dimensions'][0]): int(row['metrics'][0]['values'][0])
```

Comment:
Here's the results for a revision ID with no results:

```
{u'reports': [{u'columnHeader': {u'dimensions': [u'ga:dimension12'],
                                 u'metricHeader': {u'metricHeaderEntries': [{u'name': u'ga:users',
                                                                             u'type': u'INTEGER'}]}},
               u'data': {u'samplesReadCounts': [u'865500'],
                         u'samplingSpaceSizes': [u'2083471'],
                         u'totals': [{u'values': [u'0']}]}}]}
```

This raises the error:

```
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/home/vagrant/src/kuma/wiki/utils.py", line 108, in analytics_user_counts
    for row in response['reports'][0]['data']['rows']}
KeyError: 'rows'
```

versus a result with data:

```
{u'reports': [{u'columnHeader': {u'dimensions': [u'ga:dimension12'],
                                 u'metricHeader': {u'metricHeaderEntries': [{u'name': u'ga:users',
                                                                             u'type': u'INTEGER'}]}},
               u'data': {u'maximums': [{u'values': [u'15062']}],
                         u'minimums': [{u'values': [u'15062']}],
                         u'rowCount': 1,
                         u'rows': [{u'dimensions': [u'1068728'],
                                    u'metrics': [{u'values': [u'15062']}]}],
                         u'samplesReadCounts': [u'864180'],
                         u'samplingSpaceSizes': [u'2085651'],
                         u'totals': [{u'values': [u'15062']}]}}]}
```

which results in the expected ```{1068728: 15062}```

Maybe ``totals`` is a more consistent property?